pandas/core/series.py

```diff
@@ -2701,8 +2701,8 @@ def corr(
         if len(this) == 0:
             return np.nan
 
-        this_values = np.asarray(this._values)
-        other_values = np.asarray(other._values)
+        this_values = this.to_numpy(dtype=float, na_value=np.nan, copy=False)
```

Comment:
any chance this gets weird if we have non-numeric?

is this the right thing to do if we have NAs?

Reply:
> any chance this gets weird if we have non-numeric?

This raises for non-numeric that cannot be cast to float (same as behavior as `main` branch):

```
ser = pd.Series(["a", "b"])
ser.corr(ser)

# ValueError: could not convert string to float: 'a'
```

It allows non-numeric that can be cast to float (same behavior as `main` branch):

```
ser = pd.Series(["1", "2"])
ser.corr(ser)

# 0.9999999999999999
```
 
> is this the right thing to do if we have NAs?

NAs get dropped either way in the underlying nanops.py functions `nancorr` and `nancov`. 

FYI, this is the same approach used in `DataFrame.corr`:

https://github.com/pandas-dev/pandas/blob/0e390168407342e69886b6db7a49a281b75c9c43/pandas/core/frame.py#L10381

Reply:
> FYI, this is the same approach used in DataFrame.corr:

good point.  works for me